R/stanreg.R

```diff
@@ -73,6 +89,11 @@ stanreg <- function(object) {
     residuals <- ytmp - mu
   }
   names(eta) <- names(mu) <- names(residuals) <- ynames
+  if (family$family == "beta") {
+    eta_z <- linear_predictor(coefs_z, z, object$offset)
+    phi <- family_phi$linkinv(eta_z)
+    # residuals <- ytmp - rbeta(length(eta), mu * phi, (1 - mu) * phi) # this is not what betareg does
```

Comment:
I'm not sure why I wrote this. It seems that the way the residuals are done now match that of betareg.

```
dat <- list()
dat$N <- 200
dat$x <- rnorm(dat$N, 2, 1)
dat$z <- rnorm(dat$N, 2, 1)
dat$mu <- binomial(link = "logit")$linkinv(0.5 + 0.2*dat$x)
dat$phi <- exp(1.5 + 0.4*dat$z)
dat$y <- rbeta(dat$N, dat$mu * dat$phi, (1 - dat$mu) * dat$phi)
dat <- data.frame(dat$y, dat$x, dat$z)
colnames(dat) <- c("y", "x", "z")

check <- betareg::betareg(y ~ x | z, data = dat, link = "log", link.phi = "log")
fit <- rstanarm::stan_betareg(y ~ x | z , data = dat, link = "log", link.phi = "log",
                              chains = 4, cores = 4, algorithm = "sampling")

plot(fit$residuals)
points(check$residuals, col = "red")
```